src/main/java/com/github/tomakehurst/wiremock/stubbing/Scenario.java

```diff
@@ -72,27 +69,15 @@ public String getState() {
   }
 
   public Set<String> getPossibleStates() {
-    FluentIterable<String> requiredStates =
-        from(stubMappings)
-            .transform(
-                new Function<StubMapping, String>() {
-                  @Override
-                  public String apply(StubMapping mapping) {
-                    return mapping.getRequiredScenarioState();
-                  }
-                });
-
-    return from(stubMappings)
-        .transform(
-            new Function<StubMapping, String>() {
-              @Override
-              public String apply(StubMapping mapping) {
-                return mapping.getNewScenarioState();
-              }
-            })
-        .append(requiredStates)
-        .filter(Predicates.notNull())
-        .toSet();
+    List<String> requiredStates =
+        stubMappings.stream()
+            .map(StubMapping::getRequiredScenarioState)
+            .collect(Collectors.toList());
+
+    requiredStates.addAll(
+        stubMappings.stream().map(StubMapping::getNewScenarioState).collect(Collectors.toList()));
+
+    return requiredStates.stream().filter(Objects::nonNull).collect(Collectors.toSet());
```

Comment:
Unfortunately, small changes in the order of the final collection.

AdminApiTest -> void getScenarios()
 - assertThat(body, jsonPartEquals("scenarios[0].possibleStates", asList("2", "3", "Started")));
+ assertThat(body, jsonPartEquals("scenarios[0].possibleStates", asList("Started", "2", "3")));